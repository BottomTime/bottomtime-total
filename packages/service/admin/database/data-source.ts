import path from 'path';
import { DataSource } from 'typeorm';

export async function getDataSource(
  url: string,
  requireSsl: boolean,
): Promise<DataSource> {
  const client = new DataSource({
    type: 'postgres',
    url,
    entities: [path.resolve(__dirname, '../../src/data/**/*.entity.ts')],
    ssl: requireSsl,
  });
  return await client.initialize();
}
