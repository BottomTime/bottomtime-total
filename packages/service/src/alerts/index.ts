export * from './alert';
export * from './alerts.service';
export * from './alerts.module';
