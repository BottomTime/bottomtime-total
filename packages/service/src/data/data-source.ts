import path from 'path';
import { DataSource, DataSourceOptions } from 'typeorm';

import { Config } from '../config';

export const PostgresDataSourceOptions: DataSourceOptions = {
  type: 'postgres',
  url: Config.postgresUri,
  entities: [path.join(__dirname, './**/*.entity.ts')],
  synchronize: false,
};

export async function initDataSource(): Promise<DataSource> {
  const AppDataSource = new DataSource(PostgresDataSourceOptions);
  return await AppDataSource.initialize();
}
