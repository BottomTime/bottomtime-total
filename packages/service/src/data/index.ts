export * from './data-source';
export * from './get-query-vector';

export * from './certification.entity';
export * from './dive-site-review.entity';
export * from './dive-site.entity';
export * from './friend-request.entity';
export * from './friendship.entity';
export * from './tank.entity';
export * from './user-certification.entity';
export * from './user-oauth.entity';
export * from './user.entity';
