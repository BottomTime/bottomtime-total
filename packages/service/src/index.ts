import { Config } from './config';
import { createApp } from './create-app';
import { createLogger } from './logger';

const log = createLogger(Config.logLevel);

createApp(log)
  .then(async (app) => {
    await app.listen(Config.port);

    log.info(
      `ðŸŽ‰ Service has successfully started and is listening on port ${Config.port}. ðŸŽ‰`,
    );
  })
  .catch((error) => {
    log.fatal('[SERVICE] Failed to start service... Shutting down...', error);
    process.exit(1);
  });
