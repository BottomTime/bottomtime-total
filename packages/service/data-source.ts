import path from 'path';
import { DataSource } from 'typeorm';

import { PostgresDataSourceOptions } from './src/data/data-source';

export default new DataSource({
  ...PostgresDataSourceOptions,
  migrations: [path.join(__dirname, './migrations/*.ts')],
});
