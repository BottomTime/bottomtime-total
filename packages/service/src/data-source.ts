import 'reflect-metadata';
import { DataSourceOptions } from 'typeorm';

import { Config } from './config';
import * as Entities from './data';

export const PostgresDataSourceOptions: DataSourceOptions = {
  type: 'postgres',
  url: Config.postgresUri,
  entities: Object.values(Entities),
  synchronize: false,
};
