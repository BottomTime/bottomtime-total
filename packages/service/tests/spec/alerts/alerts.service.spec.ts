import { CreateOrUpdateAlertParamsDTO } from '@bottomtime/api';

import { Repository } from 'typeorm';

import { AlertsService } from '../../../src/alerts';
import { AlertEntity, UserEntity } from '../../../src/data';
import { dataSource } from '../../data-source';
import AlertTestData from '../../fixtures/alerts.json';
import { parseAlertJSON } from '../../utils/create-test-alert';
import { createTestUser } from '../../utils/create-test-user';

describe('Alerts Service', () => {
  let Alerts: Repository<AlertEntity>;
  let Users: Repository<UserEntity>;
  let service: AlertsService;
  let alertData: AlertEntity[];

  beforeAll(() => {
    Alerts = dataSource.getRepository(AlertEntity);
    Users = dataSource.getRepository(UserEntity);
    service = new AlertsService(Alerts);

    alertData = AlertTestData.map((data) => parseAlertJSON(data));
  });

  describe('when listing alerts', () => {
    let user: UserEntity;

    beforeAll(() => {
      user = createTestUser();
    });

    beforeEach(async () => {
      await Promise.all([Alerts.save(alertData), Users.save(user)]);
      await Promise.all(
        alertData
          .slice(0, 8)
          .map((alert) =>
            Alerts.createQueryBuilder()
              .relation('dismissals')
              .of(alert)
              .add(user),
          ),
      );
    });

    it('will return all alerts for anonymous users', async () => {
      const results = await service.listAlerts({});

      expect(results.totalCount).toBe(alertData.length);
      expect(results.alerts).toHaveLength(10);
      expect(results.alerts).toMatchSnapshot();
    });

    it('will return non-dismissed alerts by default', async () => {
      const results = await service.listAlerts({ userId: user.id });
      expect(results.totalCount).toBe(12);
      expect(results.alerts).toHaveLength(10);
      expect(results.alerts).toMatchSnapshot();
    });

    it('will return all alerts if requested', async () => {
      const results = await service.listAlerts({
        userId: user.id,
        showDismissed: true,
      });
      expect(results.totalCount).toBe(alertData.length);
      expect(results.alerts).toHaveLength(10);
      expect(results.alerts).toMatchSnapshot();
    });

    it('will allow pagination of results', async () => {
      const results = await service.listAlerts({
        userId: user.id,
        skip: 7,
        limit: 6,
      });
      expect(results.totalCount).toBe(12);
      expect(results.alerts).toHaveLength(5);
      expect(results.alerts).toMatchSnapshot();
    });

    it('will return an empty result set if no alerts can be found', async () => {
      await Alerts.delete({});
      const results = await service.listAlerts({});
      expect(results).toEqual({
        alerts: [],
        totalCount: 0,
      });
    });
  });

  describe('when retrieving a single alert', () => {
    let userData: UserEntity;
    let alert: AlertEntity;

    beforeAll(() => {
      userData = createTestUser();
      alert = alertData[0];
    });

    beforeEach(async () => {
      await Promise.all([Users.save(userData), Alerts.save(alert)]);
    });

    it('will retrieve the indicated alert', async () => {
      const result = await service.getAlert(alert.id);
      expect(result).toBeDefined();
      expect(result?.id).toBe(alert.id);
      expect(result?.icon).toBe(alert.icon);
      expect(result?.title).toBe(alert.title);
      expect(result?.message).toBe(alert.message);
      expect(result?.active).toEqual(alert.active);
      expect(result?.expires).toEqual(alert.expires);
    });

    it('will return undefined if the alert does not exist', async () => {
      const result = await service.getAlert(
        'dc8aa023-7d70-4b10-8fae-f4cfefe10c92',
      );
      expect(result).toBeUndefined();
    });
  });

  it('will create a new alert and return it', async () => {
    const options: CreateOrUpdateAlertParamsDTO = {
      active: new Date('2023-03-26T15:02:14.037Z'),
      expires: new Date('2025-04-26T15:02:14.037Z'),
      icon: 'fas fa-user',
      message: 'This is a test alert.',
      title: 'Test Alert',
    };

    const alert = await service.createAlert(options);
    expect(alert.active).toEqual(options.active);
    expect(alert.expires).toEqual(options.expires);
    expect(alert.icon).toBe(options.icon);
    expect(alert.message).toBe(options.message);
    expect(alert.title).toBe(options.title);

    const saved = await Alerts.findOneByOrFail({ id: alert.id });
    expect(saved.active).toEqual(options.active);
    expect(saved.expires).toEqual(options.expires);
    expect(saved.icon).toBe(options.icon);
    expect(saved.message).toBe(options.message);
    expect(saved.title).toBe(options.title);
  });
});
