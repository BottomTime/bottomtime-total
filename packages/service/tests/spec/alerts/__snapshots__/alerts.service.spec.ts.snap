// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alerts Service when listing alerts will allow pagination of results 1`] = `
[
  {
    "active": 1707187630097,
    "expires": 1713937542011,
    "icon": "um",
    "id": "a30d2946-362b-4181-b153-fc7394c34174",
    "message": "Cum sint facilis atque ex laboriosam veniam reiciendis at dicta. Voluptates eaque cumque nihil. Similique ex sapiente maxime eum numquam dolor quam. Nostrum vel rem sequi.
Quaerat perferendis reiciendis totam possimus. Officiis dolores ipsum enim dolorum. Officiis praesentium pariatur aperiam fugit aliquid quo neque.",
    "title": "Quasi dolore laborum odio adipisci.",
  },
  {
    "active": 1706813807206,
    "expires": 1713315627097,
    "icon": "doloremque",
    "id": "38371038-abda-410e-992c-0e94cb2fb882",
    "message": "Ducimus at eaque inventore doloremque. Nostrum ullam unde adipisci sint non labore consequatur veniam. Sapiente possimus reprehenderit veritatis ab quam accusamus a velit reiciendis. Blanditiis distinctio illum possimus. Hic eius aut voluptates qui quibusdam optio molestiae. Assumenda aperiam in iste.
Tempora deserunt accusamus alias magni perspiciatis nam provident doloremque adipisci. Expedita sunt aut accusamus reprehenderit illum voluptatem. Facilis expedita quae ab culpa commodi voluptates sit.",
    "title": "Molestiae veniam unde cumque.",
  },
  {
    "active": 1706786808628,
    "expires": 1715271790270,
    "icon": "Blues",
    "id": "8e270e6f-8e0b-47c5-8ec1-7e231b0ebca0",
    "message": "Fugiat deleniti nemo voluptas reprehenderit quia pariatur. Quaerat dicta at asperiores laudantium corporis natus aliquam. Ipsam eligendi veniam reiciendis. Laborum quo ex laborum expedita tenetur velit. Quod magnam blanditiis provident iure omnis in hic. Illum consequuntur reprehenderit voluptatibus consequatur beatae eius itaque neque.
Suscipit provident molestias odit incidunt ullam. Perferendis perspiciatis reiciendis deserunt error totam. Facere inventore libero doloremque. Odit voluptas porro ipsum ratione iste odit. Quia possimus ex ea dolore. Aperiam nulla nostrum sunt.",
    "title": "Eum eius corrupti ex.",
  },
  {
    "active": 1706784335637,
    "expires": 1714578074815,
    "icon": "Borders",
    "id": "d2b624df-0a39-40c7-be93-5b4a4f3763d0",
    "message": "Corporis eveniet odit numquam. Reprehenderit corporis assumenda delectus dolore sed. Quas esse architecto totam dolores occaecati. Consequuntur magnam quam mollitia rem nostrum. A iure nobis repellat est voluptate est natus qui.
Vitae animi minima earum minus doloribus ratione. Amet sunt corrupti assumenda alias reiciendis voluptas dolorem deserunt asperiores. Sint illo ut repellat laboriosam saepe laborum. Id culpa sed animi voluptate praesentium asperiores. Repellendus nulla consequatur perferendis dolores. Harum quia eum ratione similique iste provident nisi.",
    "title": "Nobis quos eveniet quae fugit qui iste.",
  },
  {
    "active": 1706487352621,
    "expires": 1715075767341,
    "icon": "New",
    "id": "806d4b14-367f-41c9-9840-27bf08e65d8d",
    "message": "Ducimus id quos illum eos fugiat. Illo delectus ipsa perferendis quaerat saepe consectetur. Voluptates exercitationem quia quod. Deserunt assumenda tempora vitae enim. Quibusdam nisi repellendus sint.
Tempore placeat quis. Ipsam possimus saepe. Culpa ipsum doloremque nisi eos saepe incidunt ab officiis cum. Nihil quae quibusdam. Ab maxime totam suscipit molestias a ratione asperiores. Officia accusamus illum tenetur dolores quod quisquam vitae labore.",
    "title": "Harum dolor in expedita a.",
  },
]
`;

exports[`Alerts Service when listing alerts will return all alerts for anonymous users 1`] = `
[
  {
    "active": 1711129987600,
    "expires": 1714827309699,
    "icon": "female",
    "id": "ac7f6181-aa8a-4440-884b-d7eb28d3ef92",
    "message": "Quia officiis dolore illum laudantium ea animi distinctio. Sit explicabo quis quis odio esse. Iste ratione quod omnis libero. Accusantium tempora aspernatur corporis assumenda deleniti eveniet delectus earum.
Blanditiis delectus minima maiores adipisci beatae ab. Nostrum nulla ipsa cum distinctio voluptatem blanditiis distinctio sunt earum. Distinctio ratione vero veniam. Beatae eum natus. Magni maiores veniam culpa ipsam dolorum.",
    "title": "Aperiam quia ipsum illo.",
  },
  {
    "active": 1711051129588,
    "expires": undefined,
    "icon": "gamebird",
    "id": "0ccc27a7-1597-4934-84ad-873c42997ec9",
    "message": "Tempora rem adipisci corrupti repellat facilis expedita a reprehenderit ut. Odio reprehenderit laboriosam est voluptatum cupiditate eos ipsum officia iusto. Ea deserunt natus possimus. Consectetur quis impedit quas eos.
Porro aut ducimus perspiciatis unde quibusdam fugiat nemo tenetur aliquid. Inventore facere ducimus est reiciendis. Voluptatem omnis saepe fugit nobis unde eligendi eum.",
    "title": "Laboriosam dolorem ipsam unde explicabo alias molestiae.",
  },
  {
    "active": 1709559201151,
    "expires": 1713063367964,
    "icon": "Technician",
    "id": "6691762f-b4c7-43cf-a424-b857acc0f59a",
    "message": "Tempore doloremque dignissimos ut distinctio sit ex molestiae. Tenetur tenetur totam a distinctio placeat voluptatem ipsam. Modi libero sapiente dicta.
Eligendi sit eius quis vel saepe exercitationem incidunt. Pariatur error ipsam tenetur vero facilis alias ex. Non tempora fuga dolorum voluptatum hic. Neque perspiciatis fuga voluptatem quisquam consequatur minus. Amet sit asperiores minus occaecati itaque cum.",
    "title": "A illum reiciendis rerum ratione maxime dolorem dignissimos.",
  },
  {
    "active": 1709454759818,
    "expires": 1715619194985,
    "icon": "Rap",
    "id": "38d5da43-9837-4bf2-8142-60f4a518e969",
    "message": "Voluptatibus eveniet sit tempora consequatur. Deleniti voluptatum laborum harum. Impedit repudiandae repellat doloremque excepturi dolorum unde ratione vero libero. Dolore alias ab exercitationem a voluptates quam distinctio.
Dicta nulla minima temporibus cumque nam molestiae quos. Laudantium nobis perspiciatis ipsam repellendus velit tempore adipisci quisquam aut. Optio distinctio iste tempora corporis aliquid facilis illo ipsam voluptatibus.",
    "title": "Architecto iste culpa praesentium doloribus dicta totam consectetur placeat minima.",
  },
  {
    "active": 1708686392015,
    "expires": 1715208590725,
    "icon": "Northeast",
    "id": "58721d09-c80c-4d57-a3f5-85c69d0acda2",
    "message": "Hic sed vero suscipit vitae. Hic nam nobis nemo explicabo. Quisquam quaerat ex suscipit est ducimus ex odit laboriosam aperiam. Ut ipsa quibusdam laborum nisi rem exercitationem iste.
Cumque esse sit iure ipsum. Praesentium explicabo aspernatur modi odit est officiis deleniti nemo. Veniam porro excepturi alias repellat natus pariatur. Ratione magnam aperiam temporibus assumenda necessitatibus impedit hic explicabo.",
    "title": "Repellendus velit est voluptates corporis officia quaerat cupiditate.",
  },
  {
    "active": 1708648735904,
    "expires": 1711648671516,
    "icon": "Product",
    "id": "c4d1a4dc-f273-4c19-9b11-9ceaf1c01e3d",
    "message": "Numquam sint autem. Maxime aliquid doloremque modi quidem. Minima quae voluptatibus sint fugit minima quas saepe praesentium.
Culpa illum nesciunt reprehenderit voluptatem nemo nihil vitae. Rem quos repellat laborum suscipit. Dolorum perferendis dignissimos pariatur illo illum assumenda harum. Quae ipsam amet quidem voluptatibus. Laboriosam quibusdam repudiandae ut odio soluta in corrupti possimus quibusdam.",
    "title": "Iusto iusto eos accusantium ea repellendus laborum dignissimos saepe maiores.",
  },
  {
    "active": 1708573479159,
    "expires": 1716214037597,
    "icon": "Hyundai",
    "id": "7d1c768b-f1df-4032-b822-7f01fe68ef80",
    "message": "Eius nobis consequuntur occaecati quod. Tempora velit sapiente pariatur rem iusto ipsam. Magni eos maxime earum placeat. Iure temporibus dolorum dolores minima. Consequatur nostrum quod occaecati veritatis iure. Saepe accusantium corrupti dolorum odit cum.
Quis eum quae aliquid velit repellat nam blanditiis. Expedita pariatur aperiam impedit fugit quod distinctio voluptas. Beatae fugiat fugiat. Laboriosam omnis labore modi modi. Commodi eos corrupti id est ipsum animi laudantium quidem. Quos odio deserunt dolor aut voluptatum.",
    "title": "Voluptatem eius officiis iure dolorem quam ratione eos recusandae in.",
  },
  {
    "active": 1708527238427,
    "expires": 1713387033926,
    "icon": "Reggae",
    "id": "ea9e3a54-ef36-4d2e-aa92-a4c932ca45d8",
    "message": "Numquam quo odit ex aspernatur aliquam totam suscipit perspiciatis iste. Iure sequi libero quis eaque rerum eaque. Facere laboriosam accusantium in fugit. Soluta dolore voluptatibus. Ipsum aliquam perferendis rem reiciendis tenetur.
Praesentium saepe consectetur autem vero labore provident tempora provident. Non quaerat voluptatem repellendus. Illum error quo neque. Necessitatibus quae labore.",
    "title": "Quod dolore debitis laboriosam numquam similique.",
  },
  {
    "active": 1708067545064,
    "expires": undefined,
    "icon": "plum",
    "id": "6a579869-fe57-4b7c-89e5-13b839b28863",
    "message": "Quam modi quas minima deleniti minus veniam quos quasi hic. Inventore dolorem quibusdam dolore vel illo. Iure odio vero nulla. Ad neque sapiente. Itaque ipsa aspernatur assumenda tempore tenetur impedit. Deserunt quos quisquam exercitationem beatae quasi eum enim.
Incidunt consectetur voluptas. Beatae incidunt maiores ipsa eius ut modi dicta officiis. Voluptates quibusdam assumenda et voluptates culpa.",
    "title": "Cupiditate blanditiis vero dolores quisquam voluptas beatae nostrum possimus alias.",
  },
  {
    "active": 1707990504875,
    "expires": 1711637105532,
    "icon": "evenly",
    "id": "41d51fe5-7de6-4b5b-85e3-ec7e169e9dfc",
    "message": "Libero facilis officia ducimus vero dicta. Nulla illum corporis odio amet impedit ipsa. Aperiam est pariatur. Fugit eveniet magnam nostrum quam temporibus fugit. Autem dicta cupiditate non. Facere consequatur tenetur quam asperiores dolore nam magni.
Aliquam aspernatur ullam odit veritatis sapiente. Magnam dolor nihil consectetur ut modi natus neque. Doloremque facere quasi soluta tempore accusantium amet at. Excepturi nulla ipsa deserunt commodi.",
    "title": "Expedita eius fuga occaecati.",
  },
]
`;

exports[`Alerts Service when listing alerts will return all alerts if requested 1`] = `
[
  {
    "active": 1711129987600,
    "expires": 1714827309699,
    "icon": "female",
    "id": "ac7f6181-aa8a-4440-884b-d7eb28d3ef92",
    "message": "Quia officiis dolore illum laudantium ea animi distinctio. Sit explicabo quis quis odio esse. Iste ratione quod omnis libero. Accusantium tempora aspernatur corporis assumenda deleniti eveniet delectus earum.
Blanditiis delectus minima maiores adipisci beatae ab. Nostrum nulla ipsa cum distinctio voluptatem blanditiis distinctio sunt earum. Distinctio ratione vero veniam. Beatae eum natus. Magni maiores veniam culpa ipsam dolorum.",
    "title": "Aperiam quia ipsum illo.",
  },
  {
    "active": 1711051129588,
    "expires": undefined,
    "icon": "gamebird",
    "id": "0ccc27a7-1597-4934-84ad-873c42997ec9",
    "message": "Tempora rem adipisci corrupti repellat facilis expedita a reprehenderit ut. Odio reprehenderit laboriosam est voluptatum cupiditate eos ipsum officia iusto. Ea deserunt natus possimus. Consectetur quis impedit quas eos.
Porro aut ducimus perspiciatis unde quibusdam fugiat nemo tenetur aliquid. Inventore facere ducimus est reiciendis. Voluptatem omnis saepe fugit nobis unde eligendi eum.",
    "title": "Laboriosam dolorem ipsam unde explicabo alias molestiae.",
  },
  {
    "active": 1709559201151,
    "expires": 1713063367964,
    "icon": "Technician",
    "id": "6691762f-b4c7-43cf-a424-b857acc0f59a",
    "message": "Tempore doloremque dignissimos ut distinctio sit ex molestiae. Tenetur tenetur totam a distinctio placeat voluptatem ipsam. Modi libero sapiente dicta.
Eligendi sit eius quis vel saepe exercitationem incidunt. Pariatur error ipsam tenetur vero facilis alias ex. Non tempora fuga dolorum voluptatum hic. Neque perspiciatis fuga voluptatem quisquam consequatur minus. Amet sit asperiores minus occaecati itaque cum.",
    "title": "A illum reiciendis rerum ratione maxime dolorem dignissimos.",
  },
  {
    "active": 1709454759818,
    "expires": 1715619194985,
    "icon": "Rap",
    "id": "38d5da43-9837-4bf2-8142-60f4a518e969",
    "message": "Voluptatibus eveniet sit tempora consequatur. Deleniti voluptatum laborum harum. Impedit repudiandae repellat doloremque excepturi dolorum unde ratione vero libero. Dolore alias ab exercitationem a voluptates quam distinctio.
Dicta nulla minima temporibus cumque nam molestiae quos. Laudantium nobis perspiciatis ipsam repellendus velit tempore adipisci quisquam aut. Optio distinctio iste tempora corporis aliquid facilis illo ipsam voluptatibus.",
    "title": "Architecto iste culpa praesentium doloribus dicta totam consectetur placeat minima.",
  },
  {
    "active": 1708686392015,
    "expires": 1715208590725,
    "icon": "Northeast",
    "id": "58721d09-c80c-4d57-a3f5-85c69d0acda2",
    "message": "Hic sed vero suscipit vitae. Hic nam nobis nemo explicabo. Quisquam quaerat ex suscipit est ducimus ex odit laboriosam aperiam. Ut ipsa quibusdam laborum nisi rem exercitationem iste.
Cumque esse sit iure ipsum. Praesentium explicabo aspernatur modi odit est officiis deleniti nemo. Veniam porro excepturi alias repellat natus pariatur. Ratione magnam aperiam temporibus assumenda necessitatibus impedit hic explicabo.",
    "title": "Repellendus velit est voluptates corporis officia quaerat cupiditate.",
  },
  {
    "active": 1708648735904,
    "expires": 1711648671516,
    "icon": "Product",
    "id": "c4d1a4dc-f273-4c19-9b11-9ceaf1c01e3d",
    "message": "Numquam sint autem. Maxime aliquid doloremque modi quidem. Minima quae voluptatibus sint fugit minima quas saepe praesentium.
Culpa illum nesciunt reprehenderit voluptatem nemo nihil vitae. Rem quos repellat laborum suscipit. Dolorum perferendis dignissimos pariatur illo illum assumenda harum. Quae ipsam amet quidem voluptatibus. Laboriosam quibusdam repudiandae ut odio soluta in corrupti possimus quibusdam.",
    "title": "Iusto iusto eos accusantium ea repellendus laborum dignissimos saepe maiores.",
  },
  {
    "active": 1708573479159,
    "expires": 1716214037597,
    "icon": "Hyundai",
    "id": "7d1c768b-f1df-4032-b822-7f01fe68ef80",
    "message": "Eius nobis consequuntur occaecati quod. Tempora velit sapiente pariatur rem iusto ipsam. Magni eos maxime earum placeat. Iure temporibus dolorum dolores minima. Consequatur nostrum quod occaecati veritatis iure. Saepe accusantium corrupti dolorum odit cum.
Quis eum quae aliquid velit repellat nam blanditiis. Expedita pariatur aperiam impedit fugit quod distinctio voluptas. Beatae fugiat fugiat. Laboriosam omnis labore modi modi. Commodi eos corrupti id est ipsum animi laudantium quidem. Quos odio deserunt dolor aut voluptatum.",
    "title": "Voluptatem eius officiis iure dolorem quam ratione eos recusandae in.",
  },
  {
    "active": 1708527238427,
    "expires": 1713387033926,
    "icon": "Reggae",
    "id": "ea9e3a54-ef36-4d2e-aa92-a4c932ca45d8",
    "message": "Numquam quo odit ex aspernatur aliquam totam suscipit perspiciatis iste. Iure sequi libero quis eaque rerum eaque. Facere laboriosam accusantium in fugit. Soluta dolore voluptatibus. Ipsum aliquam perferendis rem reiciendis tenetur.
Praesentium saepe consectetur autem vero labore provident tempora provident. Non quaerat voluptatem repellendus. Illum error quo neque. Necessitatibus quae labore.",
    "title": "Quod dolore debitis laboriosam numquam similique.",
  },
  {
    "active": 1708067545064,
    "expires": undefined,
    "icon": "plum",
    "id": "6a579869-fe57-4b7c-89e5-13b839b28863",
    "message": "Quam modi quas minima deleniti minus veniam quos quasi hic. Inventore dolorem quibusdam dolore vel illo. Iure odio vero nulla. Ad neque sapiente. Itaque ipsa aspernatur assumenda tempore tenetur impedit. Deserunt quos quisquam exercitationem beatae quasi eum enim.
Incidunt consectetur voluptas. Beatae incidunt maiores ipsa eius ut modi dicta officiis. Voluptates quibusdam assumenda et voluptates culpa.",
    "title": "Cupiditate blanditiis vero dolores quisquam voluptas beatae nostrum possimus alias.",
  },
  {
    "active": 1707990504875,
    "expires": 1711637105532,
    "icon": "evenly",
    "id": "41d51fe5-7de6-4b5b-85e3-ec7e169e9dfc",
    "message": "Libero facilis officia ducimus vero dicta. Nulla illum corporis odio amet impedit ipsa. Aperiam est pariatur. Fugit eveniet magnam nostrum quam temporibus fugit. Autem dicta cupiditate non. Facere consequatur tenetur quam asperiores dolore nam magni.
Aliquam aspernatur ullam odit veritatis sapiente. Magnam dolor nihil consectetur ut modi natus neque. Doloremque facere quasi soluta tempore accusantium amet at. Excepturi nulla ipsa deserunt commodi.",
    "title": "Expedita eius fuga occaecati.",
  },
]
`;

exports[`Alerts Service when listing alerts will return non-dismissed alerts by default 1`] = `
[
  {
    "active": 1711129987600,
    "expires": 1714827309699,
    "icon": "female",
    "id": "ac7f6181-aa8a-4440-884b-d7eb28d3ef92",
    "message": "Quia officiis dolore illum laudantium ea animi distinctio. Sit explicabo quis quis odio esse. Iste ratione quod omnis libero. Accusantium tempora aspernatur corporis assumenda deleniti eveniet delectus earum.
Blanditiis delectus minima maiores adipisci beatae ab. Nostrum nulla ipsa cum distinctio voluptatem blanditiis distinctio sunt earum. Distinctio ratione vero veniam. Beatae eum natus. Magni maiores veniam culpa ipsam dolorum.",
    "title": "Aperiam quia ipsum illo.",
  },
  {
    "active": 1711051129588,
    "expires": undefined,
    "icon": "gamebird",
    "id": "0ccc27a7-1597-4934-84ad-873c42997ec9",
    "message": "Tempora rem adipisci corrupti repellat facilis expedita a reprehenderit ut. Odio reprehenderit laboriosam est voluptatum cupiditate eos ipsum officia iusto. Ea deserunt natus possimus. Consectetur quis impedit quas eos.
Porro aut ducimus perspiciatis unde quibusdam fugiat nemo tenetur aliquid. Inventore facere ducimus est reiciendis. Voluptatem omnis saepe fugit nobis unde eligendi eum.",
    "title": "Laboriosam dolorem ipsam unde explicabo alias molestiae.",
  },
  {
    "active": 1709454759818,
    "expires": 1715619194985,
    "icon": "Rap",
    "id": "38d5da43-9837-4bf2-8142-60f4a518e969",
    "message": "Voluptatibus eveniet sit tempora consequatur. Deleniti voluptatum laborum harum. Impedit repudiandae repellat doloremque excepturi dolorum unde ratione vero libero. Dolore alias ab exercitationem a voluptates quam distinctio.
Dicta nulla minima temporibus cumque nam molestiae quos. Laudantium nobis perspiciatis ipsam repellendus velit tempore adipisci quisquam aut. Optio distinctio iste tempora corporis aliquid facilis illo ipsam voluptatibus.",
    "title": "Architecto iste culpa praesentium doloribus dicta totam consectetur placeat minima.",
  },
  {
    "active": 1708686392015,
    "expires": 1715208590725,
    "icon": "Northeast",
    "id": "58721d09-c80c-4d57-a3f5-85c69d0acda2",
    "message": "Hic sed vero suscipit vitae. Hic nam nobis nemo explicabo. Quisquam quaerat ex suscipit est ducimus ex odit laboriosam aperiam. Ut ipsa quibusdam laborum nisi rem exercitationem iste.
Cumque esse sit iure ipsum. Praesentium explicabo aspernatur modi odit est officiis deleniti nemo. Veniam porro excepturi alias repellat natus pariatur. Ratione magnam aperiam temporibus assumenda necessitatibus impedit hic explicabo.",
    "title": "Repellendus velit est voluptates corporis officia quaerat cupiditate.",
  },
  {
    "active": 1708573479159,
    "expires": 1716214037597,
    "icon": "Hyundai",
    "id": "7d1c768b-f1df-4032-b822-7f01fe68ef80",
    "message": "Eius nobis consequuntur occaecati quod. Tempora velit sapiente pariatur rem iusto ipsam. Magni eos maxime earum placeat. Iure temporibus dolorum dolores minima. Consequatur nostrum quod occaecati veritatis iure. Saepe accusantium corrupti dolorum odit cum.
Quis eum quae aliquid velit repellat nam blanditiis. Expedita pariatur aperiam impedit fugit quod distinctio voluptas. Beatae fugiat fugiat. Laboriosam omnis labore modi modi. Commodi eos corrupti id est ipsum animi laudantium quidem. Quos odio deserunt dolor aut voluptatum.",
    "title": "Voluptatem eius officiis iure dolorem quam ratione eos recusandae in.",
  },
  {
    "active": 1707990504875,
    "expires": 1711637105532,
    "icon": "evenly",
    "id": "41d51fe5-7de6-4b5b-85e3-ec7e169e9dfc",
    "message": "Libero facilis officia ducimus vero dicta. Nulla illum corporis odio amet impedit ipsa. Aperiam est pariatur. Fugit eveniet magnam nostrum quam temporibus fugit. Autem dicta cupiditate non. Facere consequatur tenetur quam asperiores dolore nam magni.
Aliquam aspernatur ullam odit veritatis sapiente. Magnam dolor nihil consectetur ut modi natus neque. Doloremque facere quasi soluta tempore accusantium amet at. Excepturi nulla ipsa deserunt commodi.",
    "title": "Expedita eius fuga occaecati.",
  },
  {
    "active": 1707927881212,
    "expires": 1715921092601,
    "icon": "Jupiter",
    "id": "8176fc01-d18d-4b0a-8941-84fc82798aef",
    "message": "Accusamus temporibus sed omnis quo. Delectus facilis sit molestias aliquam hic. Debitis quisquam quis doloribus voluptatem nisi ex repellat. Aspernatur omnis fugit amet id est officia. Optio vel at quos ipsum vero praesentium officia est. Veritatis porro consequuntur consectetur autem libero praesentium.
Quod voluptates voluptas. Quod sit consectetur voluptates sint. Modi ipsum quisquam. Sint ut sed ducimus nisi dolorum non. Reprehenderit esse doloribus ad nam adipisci suscipit. Corporis quas fugit corrupti.",
    "title": "Ratione occaecati esse veritatis.",
  },
  {
    "active": 1707187630097,
    "expires": 1713937542011,
    "icon": "um",
    "id": "a30d2946-362b-4181-b153-fc7394c34174",
    "message": "Cum sint facilis atque ex laboriosam veniam reiciendis at dicta. Voluptates eaque cumque nihil. Similique ex sapiente maxime eum numquam dolor quam. Nostrum vel rem sequi.
Quaerat perferendis reiciendis totam possimus. Officiis dolores ipsum enim dolorum. Officiis praesentium pariatur aperiam fugit aliquid quo neque.",
    "title": "Quasi dolore laborum odio adipisci.",
  },
  {
    "active": 1706813807206,
    "expires": 1713315627097,
    "icon": "doloremque",
    "id": "38371038-abda-410e-992c-0e94cb2fb882",
    "message": "Ducimus at eaque inventore doloremque. Nostrum ullam unde adipisci sint non labore consequatur veniam. Sapiente possimus reprehenderit veritatis ab quam accusamus a velit reiciendis. Blanditiis distinctio illum possimus. Hic eius aut voluptates qui quibusdam optio molestiae. Assumenda aperiam in iste.
Tempora deserunt accusamus alias magni perspiciatis nam provident doloremque adipisci. Expedita sunt aut accusamus reprehenderit illum voluptatem. Facilis expedita quae ab culpa commodi voluptates sit.",
    "title": "Molestiae veniam unde cumque.",
  },
  {
    "active": 1706786808628,
    "expires": 1715271790270,
    "icon": "Blues",
    "id": "8e270e6f-8e0b-47c5-8ec1-7e231b0ebca0",
    "message": "Fugiat deleniti nemo voluptas reprehenderit quia pariatur. Quaerat dicta at asperiores laudantium corporis natus aliquam. Ipsam eligendi veniam reiciendis. Laborum quo ex laborum expedita tenetur velit. Quod magnam blanditiis provident iure omnis in hic. Illum consequuntur reprehenderit voluptatibus consequatur beatae eius itaque neque.
Suscipit provident molestias odit incidunt ullam. Perferendis perspiciatis reiciendis deserunt error totam. Facere inventore libero doloremque. Odit voluptas porro ipsum ratione iste odit. Quia possimus ex ea dolore. Aperiam nulla nostrum sunt.",
    "title": "Eum eius corrupti ex.",
  },
]
`;
